src/adapter/src/client.rs

```diff
@@ -485,6 +488,19 @@ impl SessionClient {
             .await
     }
 
+    /// Tells the coordinator a statement has finished execution, in the cases
+    /// where we have no other reason to communicate with the coordinator.
+    pub fn retire_execute(
+        &mut self,
+        data: ExecuteContextExtra,
+        reason: StatementEndedExecutionReason,
+    ) {
+        if !data.is_trivial() {
+            let cmd = Command::RetireExecute { data, reason };
```

Comment:
I logically find it weird that in the trivial case, `reason` is ignored. Is that ok?

Reply:
Yeah, because "trivial" here means "there is nothing to log" (e.g., we are not doing statement logging at all for this statement)